File: python/heartbeat.py
import time
import json
import random
import redis

r = redis.Redis()

NODE_ID = "python-node-1"

while True:
    heartbeat = {
        "node": NODE_ID,
        "type": "heartbeat",
        "cpu": random.random(),
        "memory": random.random(),
        "timestamp": time.time()
    }

    r.publish("omega.heartbeat", json.dumps(heartbeat))

    print("🐍 HEARTBEAT:", heartbeat)

    time.sleep(3)

File: core/heartbeat.js
const { publish } = require("./event_bus");

const NODE_ID = process.argv[2] || "node-unknown";

setInterval(() => {
    const heartbeat = {
        node: NODE_ID,
        type: "heartbeat",
        cpu: Math.random().toFixed(2),
        memory: Math.random().toFixed(2),
        timestamp: Date.now()
    };

    publish("omega.heartbeat", heartbeat);

    console.log("💓 HEARTBEAT:", heartbeat);
}, 3000);
